bluetooth: host a2dp : Fix issue about re_operate after cmd works fail

CC0918 opened the pull request:
The cases as follow:
case 1: if peer device reject close cmd, then send close cmd again, it works fail because the ep->stream is NULL.
case 2: if local device set configuration and peer device get configuration, it works fail because local device does not save configuration.

MarkWangChinese commented on subsys/bluetooth/host/classic/avdtp.c:
```diff
@@ -1658,6 +1662,7 @@ static int avdtp_process_configure_command(struct bt_avdtp *session, uint8_t cmd
 		return -EINVAL;
 	}
 
+	param->sep->session = session;
```
Put after `avdtp_create_pdu`.

CC0918 replied:
Done

lylezhu2012 commented on subsys/bluetooth/host/classic/avdtp.c:
```diff
@@ -1664,6 +1668,7 @@ static int avdtp_process_configure_command(struct bt_avdtp *session, uint8_t cmd
 		return -ENOMEM;
 	}
 
+	param->sep->session = session;
```
I do not think the session should be set here.

CC0918 committed:
bluetooth: host a2dp : Fix issue about re_operate after cmd works fail

The cases as follow:
case 1: if peer device reject close cmd, then send close cmd again,
it works fail because the ep->stream is NULL.
case 2: if local device set configuration and peer device get
configuration, it works fail because local device does not
save configuration.

Signed-off-by: Cheng Chang <cheng.chang@nxp.com>

gzh-terry commented on subsys/bluetooth/host/classic/a2dp.c:
```diff
@@ -798,7 +804,7 @@ static int bt_a2dp_ctrl_cb(struct bt_avdtp_req *req, bt_a2dp_rsp_cb rsp_cb, bt_a
 	stream = ep->stream;
 
-	if (clear_stream) {
+	if (clear_stream && req->status == BT_AVDTP_SUCCESS) {
```
For example, one of the scenario is when AVDTP_ABORT is processed.
Typically it means an error occured and we can no longer maintain the stream. Then `ep->stream` should be cleared regardless of the response.
